activerecord/lib/active_record/persistence.rb

```diff
@@ -621,8 +621,14 @@ def reload(options = nil)
 
     # Saves the record with the updated_at/on attributes set to the current time
     # or the time specified.
-    # Please note that no validation is performed and only the +after_touch+,
-    # +after_commit+ and +after_rollback+ callbacks are executed.
+    #
+    # * No validation is performed.
+    # * Only the +after_touch+, +after_commit+ and +after_rollback+ callbacks
+    #   are executed.
+    # * The dirty-tracking state is cleared, so unpersisted changes are no
+    #   longer tracked by methods like +changed?+, +previous_changes+,
+    #   and +changes+. Then, there is no dirty-tracking of the +touch+ itself,
```

Comment:
This sentence "The dirty-tracking state is cleared, so unpersisted changes are no longer tracked" still makes misreading, the `touch` doesn't clear *all* unpersisted changes.

Reply:
> the touch doesn't clear all unpersisted changes.

Touch only clears unpersisted changes to the affected timestamps?

Reply:
@jaredbeck Essentially, yes. It clears changes for any column you are touching. Can you reword it to "Dirty-tracking state is cleared *for touched attributes*, so ..."? See [my updated test script](https://gist.github.com/gmcgibbon/35353ef08bdc9ab5c0c4727dddc0d754).